sqlmesh/core/loader.py

```diff
@@ -626,6 +636,31 @@ def _load_environment_statements(self, macros: MacroRegistry) -> EnvironmentStat
             return EnvironmentStatements(**statements, python_env=python_env)
         return None
 
+    def _load_linting_rules(self) -> RuleSet:
+        user_rules: t.List[type[Rule]] = []
+        rule_names: t.Dict[str, Path] = {}
```

Comment:
You could consolidate these by doing:

```python
user_rules: UniqueKeyDict[str, type[Rule]] = UniqueKeyDict("rules")
```

And then at the end do `return RuleSet(user_rules.values())`. The `UniqueKeyDict` will catch any duplicates so you don't have to manually check.